File: message-api-schema-validator/src/main/java/uk/ac/jisc/rdss/tests/metadata/create/MetadataCreateRequestTest.java
package uk.ac.jisc.rdss.tests.metadata.create;

import org.junit.Test;

import uk.ac.jisc.rdss.tests.AbstractSchemaValidatorTest;

public class MetadataCreateRequestTest extends AbstractSchemaValidatorTest {

    @Override
    protected String getSchemaFileName() {
        return "messages/metadata/create/request_schema.json";
    }

    @Test
    public void validateRequest() throws Exception {
        validateJson("messages/metadata/create/request.json");
    }
}

File: message-api-schema-validator/tests/metadata/metadata_create_request_test.py
from tests.abstract_schema_validator_test import AbstractSchemaValidatorTest
from unittest import TestCase


class MetadataCreateRequestTest(AbstractSchemaValidatorTest, TestCase):
    def get_json_schema_file_name(self):
        return 'messages/body/metadata/create/request_schema.json'

    def runTest(self):
        self.validate_json('messages/body/metadata/create/request.json')
